tools/build/cdefs.h

```diff
@@ -36,6 +36,9 @@
 
 #include_next <sys/cdefs.h>
 
+/* Provide stub __kerncap when bootstrapping */
```

Comment:
I'll defer to @arichardson on this part.

Reply:
To be honest, a lot of this file should probably be identical to the real cdefs.h and support a CHERI bootstrap compiler. I'm also surprised that the `#include_next <sys/cdefs.h>` doesn't define all the stuff that's here in the correct way?

Reply:
Ah, right, because that's `tools/build/cross-build/include/common/sys/cdefs.h`. I don't know why this file even exists? But the `__kerncap` definition feels like it belongs in there, along with all the suboject bounds macros, which would render this file unnecessary.

Reply:
I don't have a strong opinion on the CHERI bootstrap compiler, if we think it's useful I can move the proper defines in there.

Reply:
We can't include the FreeBSD sys/cdefs.h since that is incompatible with system headers on macOS/Linux, so the `tools/build/cross-build/include/common/sys/cdefs.h` adds the macros that are needed.
I think __kerncap should go there.

Reply:
I don't remember why I added this file in addition to the cross-build ones, but I feel like something broke...

Reply:
Ah because it's also needed when bootstrapping on FreeBSD.